pkg/controller/migmigration/quiesce.go

```diff
@@ -207,7 +217,40 @@ func (t *Task) scaleDownDaemonSets(client k8sclient.Client) error {
 func (t *Task) suspendCronJobs(client k8sclient.Client) error {
 	trueVal := true
 	for _, ns := range t.PlanResources.MigPlan.Spec.Namespaces {
-		list := batchv1beta1.CronJobList{}
+		list := batchv1b1.CronJobList{}
+		options := k8sclient.InNamespace(ns)
+		err := client.List(
+			context.TODO(),
+			options,
+			&list)
+		if err != nil {
+			if reflect.TypeOf(err) == reflect.TypeOf(&meta.NoKindMatchError{}) {
+				return t.suspendAlphaCronJobs(client)
```

Comment:
Since the _types_ for each API version are completely different (go) structures, I don't know of a more elegant way to deal with this.  Also, there does not seem to be `errors.IsNoKindMatchError()`.

Reply:
It seems that there *ought* to be a way to determine the API version of the resource by inspection -- since the API version is available on the resource, but yeah, off the top of my head I don't know how you'd do it given the need to have a struct available ahead of time, unless you want to use one of the more generic API calls that returns an unstructured item rather than stuffing it into the API-defined struct.

Reply:
Actually, the answer may just be to pass in a runtime.Unstructured into client.List. Here's the List code:
```
func (c *client) List(ctx context.Context, opts *ListOptions, obj runtime.Object) error {
        _, ok := obj.(*unstructured.UnstructuredList)
        if ok {
                return c.unstructuredClient.List(ctx, opts, obj)
        }
        return c.typedClient.List(ctx, opts, obj)
```

Reply:
I guess you'll have to indicate the type in the options, but then you can verify API version and then convert it to the appropriate structured resource.

Reply:
I don't see any easy way to convert to the the appropriate structured resource.  The `UnstructuredList.Items` contains `Unstructured` which contains an `Objcet` of type: `map[string]interface{}`.  So, it's not like we can switch based on the API version and _cast_ to the appropriate structured type. Even if it could be _cast_, the v2alpha1 and v1beta1 (Go) types are completely different (no inheritance) so the updating (suspend=true) code would need to be duplicated.

I included a commit with an `unstructured` impl.  It's _cringy_, IMHO, but works and luckily we don't need to do this anywhere else (so far).